kolibri/plugins/hooks.py

```diff
@@ -1,59 +1,230 @@
 """
-Hooks are used for your plugin's ``hooks`` attribute, a dictionary
-with hooks you wish to use. The keys are pre-defined constants in this module or arbitrary strings. For instance::
-
-    from kolibri.plugins.base import KolibriPluginBase
-    from kolibri.plugins.hooks import NAVIGATION_POPULATE
-
-    class MyPlugin(KolibriPluginBase):
-
-        def my_navigation(self, navigation_items):
-            navigation_items.append(
-                {
-                    'menu_name': _("My list"),
-                    'menu_url': reverse_lazy("my_plugin:list_of_things")
-                }
-            )
-
-        def hooks(self):
-            return {
-                NAVIGATION_POPULATE: self.my_navigation
-            }
+Kolibri Hooks API
+-----------------
+
+What are hooks
+~~~~~~~~~~~~~~
+
+Hooks are classes that define *something* that happens. It means that you can
+"hook into a component" in Kolibri and have it do a predefined and
+parameterized *thing*. For instance, Kolibri could ask all its plugins who
+wants to add something to the user settings panel, and its then up to the
+plugins to inherit from that specific hook and feed back the parameters that
+the hook definition expects.
+
+The consequences of a hook callback are located anywhere in Kolibri. Each hook
```

Comment:
Yeah, those are the existing uses of hooks in the codebase. I think it's definitely worth looking through them, because people might not be using them as you expected? For example, see also Richard's question here - https://github.com/learningequality/kolibri/pull/59#commitcomment-17156281